Atividades/para-casa-roberta/exercicio.js

```diff
@@ -0,0 +1,165 @@
+  constructor(nome, idade, cor, castrado) {
+    this.nome = nome;
+    this.idade = idade;
+    this.cor = cor;
+    this.consultas = [];
+    this.vacinas = [];
+    this.castrado = castrado;
+  }
+}
+
+class Gato extends AnimalDomestico {
+  constructor(nome, idade, cor, castrado, externo, social) {
+    super(nome, idade, cor, castrado);
+    this.externo = externo;
+    this.social = social;
+    this.consultas = [];
+    this.vacinas = [];
+  }
+  miar() {
+    console.log("miau miau");
+  }
+  brincar() {
+    console.log(`${this.nome} use uma varinha`);
+  }
+  acariciar() {
+    if (this.social == true) {
+      console.log("ron ron ron");
+    } else {
+      console.log("schiiii");
+    }
+  }
+  vacinar(vacina) {
+    this.vacinas.push(vacina);
+    this.consultas.unshift(new Date());
+  }
+
+  castrar() {
+    if (this.castrado == true) {
+      console.log(` ${this.nome}, já está castrado.`);
+    } else {
+      console.log(`${this.nome}, foi castrado`);
+      this.consultas.unshift(new Date());
+      return (this.castrado = true);
+    }
+  }
+
+  alimentar() {
+    if (this.social == false) {
+      console.log(`${this.nome} recebe um delicioso peixe!`);
+      return this.social == true;
+    }
+  }
+}
+
+Gato.prototype.especie = "doméstico";
+
+nina = new Gato("Nina", 1.5, "frajola", true, false, false);
+
+class Cachorro extends AnimalDomestico {
+  #ferido;
+  constructor(nome, idade, cor, castrado, raca, ferido) {
+    super(nome, idade, cor, castrado);
+    this.raca = raca;
+    this.#ferido = ferido;
+    this.consultas = [];
+    this.vacinas = [];
+  }
+  latir() {
+    console.log("au au");
+  }
+
+  acariciar() {
+    console.log(`${this.nome} abana o rabinho`);
+  }
+  brincar() {
+    if (this.#ferido == true) {
+      console.log(`${this.nome} está ferido e precisa de cuidados!`);
+    }
+  }
+  castrar() {
+    if (this.castrado == true) {
+      console.log(` ${this.nome}, já está castrado.`);
+    } else {
+      console.log(`${this.nome}, foi castrado`);
+      this.consultas.unshift(new Date());
+      return (this.castrado = true);
+    }
+  }
+  vacinar(vacina) {
+    this.vacinas.push(vacina);
+    this.consultas.unshift(new Date());
+  }
+}
+
+Cachorro.prototype.especie = "doméstico";
+
+zeus = new Cachorro("zeus", 3.5, "preto", false, "labrador", true);
+
+class AnimalExotico {
+  constructor(nome, idade, cor) {
+    this.nome = nome;
+    this.idade = idade;
+    this.cor = cor;
+    this.consultas = [];
+  }
+
+  consultar() {
+    this.consultas.push(new Date());
+  }
+}
+
+class Hamster extends AnimalExotico {
+  constructor(nome, idade, cor, tipo) {
+    super(nome, idade, cor);
+    this.tipo = tipo;
+  }
+  acariciar() {
+    console.log(`${this.nome} vira uma bolinha`);
+  }
+  brincar() {
+    console.log(`${this.nome} se exercita na rodinha`);
+  }
+  alimentar() {
+    console.log(`${this.nome} come frutinhas`);
+  }
+
+  consultar() {
+    return this.brincar();
+  }
```

Comment:
No consultar, eu também tava retornando brincar, mas não tava dando certo. O que eu fiz depois foi inserir um "this.relaxado" no constructor e aí, se ele tivesse relaxado a consulta seria realizada, mas se não tivesse, mostrava que ele fugiu e pedia para brincar antes. Aí no brincar(), para fazer isso, teria que mudar o status dele para relaxado.